runners/spark/src/main/java/org/apache/beam/runners/spark/translation/TranslationUtils.java

```diff
@@ -383,6 +385,6 @@ public static void rejectStateAndTimers(DoFn<?, ?> doFn) {
    * @return true if the level is memory only
    */
   public static boolean avoidRddSerialization(StorageLevel level) {
-    return level.equals(StorageLevel.MEMORY_ONLY()) || level.equals(StorageLevel.MEMORY_ONLY_2());
```

Comment:
This is a drive-by fix for using the MEMORY_ONLY_2 storage level. The code previously assumed that no serialization was necessary, which isn't strictly true since the _2 means "replicate to other nodes" - i.e. serialize over network.

Reply:
Ok, good to be sure about this. For [reference](https://spark.apache.org/docs/latest/rdd-programming-guide.html#rdd-persistence). Maybe worth to add a comment with that link.